### PR TITLE
backport: merge bitcoin#22648, #22497, #25355, #26065, #26837, #25993, #26838, #27937, partial bitcoin#22589, #22229 (i2p support: part 2)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependent on https://github.com/dashpay/dash/pull/6034

## Breaking Changes

* With I2P connections, a new, transient address is used for each outbound connection if `-i2pacceptincoming=0`.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by knst — APPROVED** [review]
utACK eea481085c1d30c76841d3aab3ee98f1c9ad5150

**Review by UdjinM6 — APPROVED** [review]
utACK

File: doc/i2p.md [hunk 1]
```diff
@@ -47,13 +47,21 @@ information in the debug log about your I2P configuration and connections. Run
...
 
 It is possible to restrict outgoing connections in the usual way with
-`onlynet=i2p`. I2P support was added to Dash Core in version 20.0 (fall 2023)
+`onlynet=i2p`. I2P support was added to Dash Core in version 20.0 (fall-2023)
 and there may be fewer I2P peers than Tor or IP ones. Therefore, using
 `onlynet=i2p` alone (without other `onlynet=`) may make a node more susceptible
 to [Sybil attacks](https://en.dash.it/wiki/Weaknesses#Sybil_attack). Use
 `dash-cli -addrinfo` to see the number of I2P addresses known to your node.
 
-## I2P related information in Dash Core
```
Comment (PastaPastaPasta) [inline doc/i2p.md:56]: 22589: I would consider this complete; we don't have a need for the other commit.

**Review by PastaPastaPasta — APPROVED** [review]
Generally utACK eea481085c1d30c76841d3aab3ee98f1c9ad5150

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK [0a23820](https://github.com/dashpay/dash/pull/6035/commits/0a23820923c875d94c7e4c6d19659fa1d39313bb); only changed commit title / text

**Pull request merged**